### PR TITLE
Drop the string representation of Duration

### PR DESCRIPTION
**Pull request by blinsay**
We've been leaning more towards using "number of seconds" as an int/float for Durations. The string is nice, but comes with a ton of additional baggage. Dropping it lets us make the `junction-api` crate not depend on kube at all.

### No more transitive dependency nonsense

`k8s-openapi` using feature flags to select a minimum kube version has weird downstream effects. To minimize those, there's now cargo feature flags in `junction-api` that allow us to do the right thing with k8s-openapi without re-importing it as a top level dependency in every crate that depends on `junction-api`.

I tried out allowing us to support multiple kube versions. The only downside of this is that `--all-features` no longer works for any of our cargo tasks, as we have multiple incompatible features.

### xtask cleanup

Changing features was a good excuse to clean up some xtasks. There are fewer options now, and it's a bit more opinionated about how we run clippy/docs/etc.

### CODE REVIEW

File: crates/junction-api-gen/src/python.rs [hunk 1]
```diff
@@ -76,9 +76,7 @@ impl PyDef {
     fn duration() -> Self {
         PyDef::Union(PyUnion {
             name: "Duration",
-            doc: Some(
-                "A duration expressed as a number of seconds or a string like '1h30m27s42ms'",
-            ),
+            doc: Some("A duration expressed as a total number of seconds. Durations should never be negative."),
             types: vec![PyType::Str, PyType::Int, PyType::Float],
```
Comment (zephraph) [inline crates/junction-api-gen/src/python.rs:80]: I assume the intent is to simplify this type into just a `PyType::Int`. Is there anything blocking that from happening?

File: crates/junction-api/src/http.rs [hunk 2]
```diff
@@ -763,7 +763,8 @@ mod tests {
         let test_json = json!({
             "codes":[ 1, 2 ],
             "attempts": 3,
-            "backoff": "1m"
+            // NOTE: serde will happily read an int here, but Duration serializes as a float
+            "backoff": 60.0,
```
Comment (zephraph) [inline crates/junction-api/src/http.rs:767]: Should it?

Comment (blinsay) [reply crates/junction-api/src/http.rs:767]: it's the best of a bunch of mediocre options right now. serializing as an int number of seconds, we risk loosing precision.

Comment (zephraph) [reply crates/junction-api/src/http.rs:767]: We chatted about this in person. Float is being used to represent sub second durations.

**Pull request merged**